sdk/keyvault/keyvault-common/tsconfig.json

```diff
@@ -1,27 +1,11 @@
 {
+  "extends": "../../../tsconfig.package",
   "compilerOptions": {
-    "alwaysStrict": true,
-    "noImplicitAny": true,
-    "preserveConstEnums": true,
-    "sourceMap": true,
-    "inlineSources": true,
-    "newLine": "LF",
-    "target": "es5",
-    "moduleResolution": "node",
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
-    "strict": true,
-    "module": "esNext",
-    "outDir": "./dist-esm",
-    "declaration": true,
-    "declarationMap": true,
-    "importHelpers": true,
     "declarationDir": "./types",
-    "lib": ["dom", "es5", "es6", "es7", "esnext"],
-    "esModuleInterop": true,
+    "outDir": "./dist-esm",
+    "lib": ["dom", "esnext.asynciterable"],
```

Comment:
you can import asynciterable where needed with a reference comment instead:

```ts
/// <reference lib="esnext.asynciterable" />
```

Reply:
Should I do that? y/n I'm up for it, but I want to understand if that's a recommendation first

Reply:
I think it's supposed to be better. Maybe @willmtemple can agree/disagree?

Reply:
I'll update it, but I wouldn't mind knowing @willmtemple 's thoughts on this!

Reply:
In this case the simplest thing to do is to keep the `lib` entry. The difference is super subtle. If we use a `///` reference, then anyone who uses this package will get `AsyncIterable` automatically, which could be surprising and could in super rare cases break a downstream build if we switch to that.

In core-http I did some funky stuff to make it so that the `///` reference was only declared ambiently for compilation but then stubbed out for distribution (because we used types from `dom.d.ts` in the core-http API surface, but that isn't the case with Asynciterable here, we only use it for implementation internally). Since we aren't exporting AsyncIterable types, we should be using the tsconfig `lib` entry or an ambient declaration file.

I think it's possible to get down to `lib: []` here, but probably out of scope for just this issue. I wouldn't make the change to using `///` here.

Reply:
Thank you Will! I will use `lib` instead in a separate PR.